### ISSUE TITLE
[BUG] Queue message shows special characters 

### ISSUE THREAD

**Issue by mishal153** [Azure/azure-sdk-for-net#10312]
I am trying to use the Azure.Storage.Queues library to insert text message into storage queue. 
When I develop an azure function using the example in https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Queues/README.md

My queue message seems to getting its coding changed or something due to which it shows up as special characters.
The code is pretty simple:

```
QueueClient queue = new QueueClient("<connectionString>", "mishaltestqueue");
string myString = "test";
await queue.SendMessageAsync(myString);
```

The queue message I get as a result is "��-" which turns out to be , as I see in hex editor, HEX - "ef bf bd ef bf bd 2d"

Azure.Storage.Queues version 12.2.0
Functions app version tried .NET core 2.2 and 3.1

Also tried to re-encode the message to utf-8 before sending like this, but that did not help

``` 
byte[] bytes = Encoding.Default.GetBytes(myString);
myString = Encoding.UTF8.GetString(bytes); 
```

Any leads on what this might be ?

It works okay if I use WindowsAzure.Storage library instead as shown below. But we are keen on using Azure.Storage.Queues dues to its support for AD based authentication instead of Key based via DefaultAzureCredential:  
```
var storageAccount = new CloudStorageAccount(new StorageCredentials(<storageName>, <storageAccessKey>), true);
var queueClient = storageAccount.CreateCloudQueueClient();
var queue = queueClient.GetQueueReference("mishaltestqueue");
var message = new CloudQueueMessage("test");
await queue.AddMessageAsync(message); 
```

**Comment by amishra-dev** [comment]
@jaschrep-msft could you try this tomorrow and respond. Thanks.

**Comment by jaschrep-msft** [comment]
Hi @mishal153. Thank you for reporting this issue.

I'm currently looking into this behavior. Could you please specify where/how you are viewing the unexpected enqueued message text? I do not see a code sample or description of the process in your issue.

**Comment by mishal153** [comment]
Hi @jaschrep-msft  , I am using the storage explorer to view the queued message. I also tried deserializing from queue but that did not work correctly either

**Comment by jaschrep-msft** [comment]
I am seeing this behavior from storage explorer, but I don't quite understand what you mean by "deserializing from queue." Does this mean you are using the 12.2.0 to download the message as well? Because I do not see that behavior.

**Comment by jaschrep-msft** [comment]
Azure Storage Explorer currently has an [issue](https://github.com/microsoft/AzureStorageExplorer/issues/1338) that appears to be the same root cause, and it has to do with base64 encoding.

Many users opt to base64 encode their messages, as this guarantees they will not interfere in the [XML structure of the REST layer](https://docs.microsoft.com/en-us/rest/api/storageservices/put-message#request-body). Storage explorer is applying a base64 decode to queue messages that can be decoded. Since the string "text" is valid base64, it is decoding it as if it were, which results in your unexpected string. The repository has a feature request to provide a toggle between displaying raw vs "decoded" values.

You can help to validate that this is the case by observing the web traffic on your machine using something like Fiddler. When Storage Explorer checks the contents of the queue, you should see the response in Fiddler with an XML body. That body will contain a bunch of message info, including a tag "MessageText" which should contain the expected string "test" that Storage Explorer is displaying incorrectly.

This is just for your unexpected behavior in storage explorer, however. I'm unsure of what you mean by "deserializing from queue" in your previous comment, and that would likely be a different cause.

**Comment by mishal153** [comment]
@jaschrep-msft Thanks ! that helps. Yes 'text' seems to be getting caught in the mix. I tried using a more real string and it does show up fine in the storage explorer. Sorry for ambiguity around "deserializing from queue" . I am trying to invoke a [queue triggered function](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-queue-trigger?tabs=csharp#encoding) via the queue message. Turns out that for the function to trigger, the message has to be base64 encoded. Which is what the QueueClient does not do by default.
However, i can encode on my end and things do work. Thanks
